src/m365/planner/commands/task/task-list.ts

```diff
@@ -14,20 +26,193 @@ class PlannerTaskListCommand extends GraphItemsListCommand<Task> {
+    return telemetryProps;
   }
 
   public defaultProperties(): string[] | undefined {
-    return ['id', 'title', 'startDateTime', 'dueDateTime', 'completedDateTime'];
+    return ['id', 'title', 'startDateTime', 'dueDateTime', 'completedDateTime', 'planId', 'bucketId'];
   }
 
   public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
-    this
-      .getAllItems(`${this.resource}/v1.0/me/planner/tasks`, logger, true)
-      .then((): void => {
-        logger.log(this.items);
-        cb();
-      }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+    const bucketId: string | undefined = args.options.bucketId;
+    const bucketName: string | undefined = args.options.bucketName;
+    const planId: string | undefined = args.options.planId;
+    const planName: string | undefined = args.options.planName;
+
+    if (bucketId || bucketName) {
+      this
+        .getBucketId(args)
+        .then((bucketId: string): Promise<void> => this.getAllItems(`${this.resource}/v1.0/planner/buckets/${bucketId}/tasks`, logger, true))
+        .then((): void => {
+          logger.log(this.items);
+          cb();
+        }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+    }
+    else if (planId || planName) {
+      this
+        .getPlanId(args)
+        .then((planId: string): Promise<void> => this.getAllItems(`${this.resource}/v1.0/planner/plans/${planId}/tasks`, logger, true))
+        .then((): void => {
+          logger.log(this.items);
+          cb();
+        }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+    }
+    else {
+      this
+        .getAllItems(`${this.resource}/v1.0/me/planner/tasks`, logger, true)
+        .then((): void => {
+          logger.log(this.items);
+          cb();
+        }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+    }
+  }
+
+  private getBucketId(args: CommandArgs): Promise<string> {
+    if (args.options.bucketId) {
+      return Promise.resolve(args.options.bucketId);
+    }
+
+    return this
+      .getPlanId(args)
+      .then((planId: string) => {
+        const requestOptions: any = {
+          url: `${this.resource}/v1.0/planner/plans/${planId}/buckets`,
+          headers: {
+            accept: 'application/json;odata.metadata=none'
+          },
+          responseType: 'json'
+        };
+
+        return request.get<{ value: { id: string; name: string; }[] }>(requestOptions);
+      })
+      .then(response => {
+        const bucket: { id: string; name: string; } | undefined = response.value.find(val => val.name === args.options.bucketName);
+
+        if (!bucket) {
+          return Promise.reject(`The specified bucket does not exist`);
+        }
+
+        return Promise.resolve(bucket.id);
+      });
+  }
+
+  private getPlanId(args: CommandArgs): Promise<string> {
+    if (args.options.planId) {
+      return Promise.resolve(args.options.planId);
```

Comment:
Since `planId` is provided by the user, we should encode it before returning it.